### ISSUE TITLE
Volume is too small

### ISSUE THREAD

**Issue by paulyoung** [paulyoung/icfs#3]
Attempting to format the volume results in `Volume is too small`: https://github.com/rafalh/rust-fatfs/blob/50b8dec36ab664b02d7532422a164e47144eecac/src/boot_sector.rs#L596

Trying to grow the stable memory any further results in an error.

Perhaps there is a bug somewhere which is giving the impression that there isn't enough available space. The `fatfs` tests demonstrate a disk with 1MB capacity: https://github.com/rafalh/rust-fatfs/blob/48d428db7f85e4e25498e2ca64cc984ea8861425/tests/format.rs#L78

**Comment by paulyoung** [comment]
```
try_fs_geometry
total_sectors: 0
bytes_per_sector: 512
sectors_per_cluster: 1
fat_type: Fat32
root_dir_sectors: 0
fats: 2
reserved_sectors: 8

Volume is too small
```

```
try_fs_geometry
total_sectors: 0
bytes_per_sector: 512
sectors_per_cluster: 1
fat_type: Fat16
root_dir_sectors: 32
fats: 2
reserved_sectors: 1

Volume is too small
```

```
try_fs_geometry
total_sectors: 0
bytes_per_sector: 512
sectors_per_cluster: 1
fat_type: Fat12
root_dir_sectors: 32
fats: 2
reserved_sectors: 1

Volume is too small
```

```
Cannot select FAT type - unfortunate storage size
```

**Comment by paulyoung** [comment]
`total_sectors` is determined here: https://github.com/rafalh/rust-fatfs/blob/8caac4468ce1c313ff4ba7e41a3033b863330e49/src/fs.rs#L1121-L1132

**Comment by paulyoung** [comment]
I believe this is fixed by 450e4b8084bf47fb63876675fb0ffd90966cf24c.

**Comment by paulyoung** [comment]
Logging now shows:

```
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] ---------- trace: format_volume
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] bytes_per_sector: 512
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] total_bytes: 1245184
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] total_sectors_64: 2432
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] total_sectors: 2432
```

```
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] ---------- try_fs_geometry
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] total_sectors: 2432
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] bytes_per_sector: 512
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] sectors_per_cluster: 2
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] fat_type: Fat32
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] root_dir_sectors: 0
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] fats: 2
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] reserved_sectors: 8
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] Invalid FAT type
```

```
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] ---------- try_fs_geometry
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] total_sectors: 2432
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] bytes_per_sector: 512
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] sectors_per_cluster: 2
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] fat_type: Fat16
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] root_dir_sectors: 32
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] fats: 2
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] reserved_sectors: 1
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] Invalid FAT type
```

```
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] ---------- try_fs_geometry
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] total_sectors: 2432
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] bytes_per_sector: 512
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] sectors_per_cluster: 2
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] fat_type: Fat12
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] root_dir_sectors: 32
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] fats: 2
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] reserved_sectors: 1
```

```
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] Panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: Io(Custom { kind: Other, error: "Attempt to read beyond available memory" }) }', src/fat/lib.rs:12:13
```
